### PR TITLE
fix(client-app): global style updates

### PR DESCRIPTION
**Pull request by antlio**
this pr favor scalar-app parent class over layer following #2135.

⚠️ current `@scalar/themes` tailwind preset config is breaking the usage of responsive modifiers + some utilities, this pr is not to be merge before including an update on the color utility namings.

### CODE REVIEW

**Comment by amritk** [comment]
Ready for review, I didn't fix the popups and menus and what not since we aren't using them on the modal. Can deal with those later. But everything else should be good to go. The font-size for xxs has grown `11px` -> `12px` so there is a  slight difference there

File: packages/client-app/src/assets/tailwind.css [hunk 1]
```diff
@@ -1,18 +1,78 @@
...
+      --tw-translate-x: -50%;
+      top: 50%;
+      --tw-translate-y: -50%;
+      transform: translate(var(--tw-translate-x), var(--tw-translate-y))
+        rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y))
+        scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
+    }
   }
+
+  text-rendering: optimizeLegibility;
```
Comment (amritk) [inline packages/client-app/src/assets/tailwind.css:70]: @hwkr do we need these here? Are these client specific or should they come from the main reset?

File: packages/client-app/src/components/DataTable/DataTableCell.vue [hunk 2]
```diff
@@ -11,7 +11,7 @@ withDefaults(
 <template>
   <component
     :is="is"
-    class="h-8 min-w-0 border-b-1/2 border-r-1/2 border-r-[inherit] text-sm last:border-r-0 group-last:border-b-transparent w-fit min-w-full has-[input[type=checkbox]]:w-8 has-[input[type=checkbox]]:min-w-8"
+    class="h-8 border-b-1/2 border-r-1/2 text-sm last:border-r-0 group-last:border-b-transparent"
```
Comment (antlio) [inline packages/client-app/src/components/DataTable/DataTableCell.vue:14]: does this maintain square on checkbox cell + hover?

File: packages/client-app/src/components/DataTable/DataTableInput.vue [hunk 3]
```diff
@@ -94,3 +96,10 @@ const handleDropdownMouseUp = () => {
       @select="handleSelect" />
   </DataTableCell>
 </template>
+
+<style scoped>
+/* Tailwind placeholder is busted */
+input::placeholder {
```
Comment (antlio) [inline packages/client-app/src/components/DataTable/DataTableInput.vue:102]: do we style it already from the theme reset style?

File: packages/components/src/components/ScalarFloating/ScalarFloating.vue [hunk 4]
```diff
@@ -76,15 +76,17 @@ const { floatingStyles, middlewareData } = useFloating(targetRef, floatingRef, {
...
-    <div
-      ref="floatingRef"
-      class="relative z-overlay"
-      :style="floatingStyles">
-      <slot
-        :data="middlewareData"
-        :height="targetHeight"
-        name="floating"
-        :width="targetWidth" />
+    <div class="scalar-app">
```
Comment (antlio) [inline packages/components/src/components/ScalarFloating/ScalarFloating.vue:79]: wondering if using scalar-components would make sense here or not

File: packages/client-app/src/assets/tailwind.css (hunk 1 above)
Comment (hwkr) [reply packages/client-app/src/assets/tailwind.css]: That's a great question, I'd just do whatever the [Tailwind preflight](https://github.com/tailwindlabs/tailwindcss/blob/next/packages/tailwindcss/preflight.css) does.

File: packages/client-app/src/components/DataTable/DataTableInput.vue (hunk 3 above)
Comment (hwkr) [reply packages/client-app/src/components/DataTable/DataTableInput.vue:102]: I think so https://github.com/scalar/scalar/blob/main/packages/themes/src/reset.css#L62

File: packages/components/src/components/ScalarFloating/ScalarFloating.vue (hunk 4 above)
Comment (hwkr) [reply packages/components/src/components/ScalarFloating/ScalarFloating.vue:79]: I could but we would then have to add more classes to the reset and stuff (probably not worth it).

File: packages/client-app/index.html [hunk 5]
```diff
@@ -26,7 +26,7 @@
   <body>
     <div
       id="app"
-      class="app"></div>
+      class="app scalar-app scalar-client"></div>
```
Comment (hwkr) [inline packages/client-app/index.html:29]: Do we want a class of `.app` at all?

File: packages/client-app/src/components/DataTable/DataTableCell.vue (hunk 2 above)
Comment (hwkr) [reply packages/client-app/src/components/DataTable/DataTableCell.vue:14]: Seems to? 

Google Chrome-2024-06-19-16-30-26@2x

File: packages/client-app/src/components/DataTable/DataTableInput.vue (hunk 3 above)
Comment (amritk) [reply packages/client-app/src/components/DataTable/DataTableInput.vue:102]: yea but didn't show up. You wanna work ur magic brynn? or we can just leave it, not the biggest thing 🤷🏾

File: packages/client-app/index.html (hunk 5 above)
Comment (amritk) [reply packages/client-app/index.html:29]: not really, don't think it does anything

File: packages/components/src/components/ScalarFloating/ScalarFloating.vue (hunk 4 above)
Comment (amritk) [reply packages/components/src/components/ScalarFloating/ScalarFloating.vue:79]: the system currently works!

**Review by marclave — APPROVED** [review]
tested and works :)

File: packages/client-app/src/components/DataTable/DataTableInput.vue (hunk 3 above)
Comment (hwkr) [reply packages/client-app/src/components/DataTable/DataTableInput.vue:102]: Oh weird, I'll look

File: packages/client-app/src/components/DataTable/DataTableCell.vue (hunk 2 above)
Comment (antlio) [reply packages/client-app/src/components/DataTable/DataTableCell.vue:14]: it's not well squared anymore, will add it back!

**Pull request merged**